telegram/src/main/kotlin/com/github/kotlintelegrambot/updater/Looper.kt

```diff
@@ -12,24 +17,23 @@ internal interface Looper {
  * intended to run the loop in a different thread). The loop will stop if the thread running the
  * loop is interrupted or in the next iteration after the [quit] method is called.
  */
-internal class ExecutorLooper(
-    private val loopExecutor: Executor,
-) : Looper {
+internal class ExecutorLooper(ioDispatcher: CoroutineDispatcher) : Looper {
 
-    @Volatile private var isLooping = false
+    private val scope: CoroutineScope = CoroutineScope(ioDispatcher)
+    private var job: Job? = null
```

Comment:
Similar comments to the ones in the `Dispatcher` class:

* We may want to make `job` volatile for memory consistency because we can't guarantee `loop` and `quit` are going to be called from the same thead, and both are accessing `job`.
* I also wonder about the `Dispatcher` we should be using here. In this case, this is actually used to run the API operations so it might make sense for it to be `Dispatchers.IO`.

Reply:
I am thinking and, as far as we cancel the job just before a new job is started, we shouldn't have memory problems. A job can be called to `cancel()` multiple times.

Reply:
The memory consistency point was about memory visibility, sorry I may have made it confusing.

As we don't know or control what thread a consumer is calling `loop` and `quit` from, we could have thread A calling `loop` and thread B calling `quit`. `loop` writes `job` and `quit` reads it, and without additional synchronization, it's not guaranteed that thread B will be able to see the value that thread A wrote to `job`. That would be solved by making `job` volatile. Does this make sense?

Reply:
Ok, make sense. Let me update it.